File: frontend/lib/models/restaurant.dart
class Restaurant {
  final int id;
  final String name;
  final String address;
  final double? lat;
  final double? lng;
  final String? category;
  final String? phone;
  final bool isKonapay;
  final bool isMobeom;
  final List<String> tags;
  final double? rating;
  final int reviewCount;
  final String geocodeStatus;
  final double? distanceKm;

  const Restaurant({
    required this.id,
    required this.name,
    required this.address,
    this.lat,
    this.lng,
    this.category,
    this.phone,
    required this.isKonapay,
    required this.isMobeom,
    required this.tags,
    this.rating,
    this.reviewCount = 0,
    this.geocodeStatus = 'verified',
    this.distanceKm,
  });

  factory Restaurant.fromJson(Map<String, dynamic> json) => Restaurant(
        id: json['id'],
        name: json['name'],
        address: json['address'] ?? '',
        lat: json['lat'] != null ? (json['lat'] as num).toDouble() : null,
        lng: json['lng'] != null ? (json['lng'] as num).toDouble() : null,
        category: json['category'],
        phone: json['phone'],
        isKonapay: json['is_konapay'] ?? false,
        isMobeom: json['is_mobeom'] ?? false,
        tags: List<String>.from(json['tags'] ?? []),
        rating: json['avg_rating'] != null
            ? (json['avg_rating'] as num).toDouble()
            : null,
        reviewCount: json['review_count'] ?? 0,
        geocodeStatus: json['geocode_status'] as String? ?? 'verified',
        distanceKm: json['distance_km'] != null
            ? (json['distance_km'] as num).toDouble()
            : null,
      );
}

File: backend/app/schemas/restaurant.py
from pydantic import BaseModel, Field


class RestaurantResponse(BaseModel):
    id: int
    name: str
    address: str
    phone: str | None
    lat: float
    lng: float
    category: str | None
    is_konapay: bool
    is_mobeom: bool
    tags: list[str] | None  # 현재 비어 있음. 식사평 기반으로 채울 자리

    # 지도 카드에 "★ 4.6 (122)" 로 쓰인다.
    # 별점을 안 남긴 식사평도 있어서 평균은 별점이 있는 것만, 개수는 전체를 센다.
    avg_rating: float | None = None
    review_count: int = 0

    # 좌표 출처. 프론트가 정확도에 따라 표시를 달리하고 싶을 때 쓴다.
    geocode_status: str

    # lat/lng 을 넘겨 조회한 경우에만 채워진다.
    distance_km: float | None = None

    model_config = {"from_attributes": True}


class RestaurantListResponse(BaseModel):
    total: int = Field(description="필터 조건에 맞는 전체 건수")
    items: list[RestaurantResponse]
